users/userRouter.js

```diff
@@ -1,47 +1,163 @@
 const express = require('express');
+const users = require('../users/userDb');
+const posts = require('../posts/postRouter');
 
 const router = express.Router();
 
-router.post('/', (req, res) => {
-  // do your magic!
+router.post('/', validateUser(), async (req, res) => {
+  const { newUser } = req.body.name;
+  users.insert({ newUser })
+  .then(newUser => {
+    res.status(201).json(newUser)
+  })
+  .catch(err => {
+    res.status(500).json({
+      errorMessage: "Error saving user to the database."
+    })
+  });
 });
 
-router.post('/:id/posts', (req, res) => {
-  // do your magic!
+router.post('/:id/posts', validateUserId(), validatePost(), (req, res) => {
+  const newPost = {
+    text: req.body.text,
+    user_id: req.params.id
+  }
+
+  posts.insert(newPost)
+    .then(newPost => {
+      res.status(201).json(newPost)
```

Comment:
insert returns an id.  Instead of `newPost` you should name it `newPostId` and then chain `getById` to return the updated resource